actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,31 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :@variants
+
+        def initialize(variants = [])
```

Comment:
If supporting `==` operator, perhaps cast this to `Set` for order-insensitive comparison.